irohad/ametsuchi/impl/postgres_command_executor.cpp

```diff
@@ -264,6 +265,31 @@ namespace {
     return query;
   }
 
+  std::string checkAccountDomainRoleOrGlobalRolePermission(
+      shared_model::interface::permissions::Role global_role,
+      shared_model::interface::permissions::Role domain_role,
+      const shared_model::interface::types::AccountIdType &creator_id,
+      const shared_model::interface::types::AccountIdType
+          &id_with_target_domain) {
+    std::string query = (boost::format(R"(WITH
+          has_global_role_perm AS (%1%),
+          has_domain_role_perm AS (%2%)
+          SELECT CASE
+                           WHEN (SELECT * FROM has_global_role_perm) THEN true
+                           WHEN ((split_part(%3%, '@', 2) = split_part(%4%, '@', 2))
```

Comment:
I think not this WHEN never used. You can change the type of id_with_target_domain to AssetIdType